apps/settings/test/unit/panels/call_barring/panel_test.js

```diff
@@ -0,0 +1,620 @@
+
+  var _mobileConnection,
+      _serviceClass;
+
+  var baocElement,
+      boicElement,
+      boicExhcElement,
+      baicElement,
+      baicRElement;
+
+  suiteSetup(function() {
+    realDsdsSettings = window.DsdsSettings;
+    window.DsdsSettings = {
+      getIccCardIndexForCallSettings: function() {
+        return 0;
+      }
+    };
+
+    realL10n = navigator.mozL10n;
+    navigator.mozL10n = MockL10n;
+
+    realMozMobileConnections = navigator.mozMobileConnections;
+    navigator.mozMobileConnections = MockNavigatorMozMobileConnections;
+
+    _mobileConnection = MockNavigatorMozMobileConnections[0];
+    _serviceClass = _mobileConnection.ICC_SERVICE_CLASS_VOICE;
+  });
+
+  suiteTeardown(function() {
+    window.DsdsSettings = realDsdsSettings;
+    navigator.mozL10n = realL10n;
+    navigator.mozMobileConnections = realMozMobileConnections;
+  });
+
+  function resetHTML() {
+    document.body.innerHTML = '';
+    loadBodyHTML('_call_barring.html');
+
+    baocElement = document.getElementById('li-cb-baoc');
+    boicElement = document.getElementById('li-cb-boic');
+    boicExhcElement = document.getElementById('li-cb-boicExhc');
+    baicElement = document.getElementById('li-cb-baic');
+    baicRElement = document.getElementById('li-cb-baicR');
+  }
+
+  function isItemDisabled(element) {
+    // getAttribute returns a 'string', asserts errors if not a boolean
+    return element.getAttribute('aria-disabled') === 'true';
+  }
+  function isItemChecked(element) {
+    return element.querySelector('input').checked || false;
+  }
+
+  setup(function(done) {
+    // Create a new requirejs context
+    var requireCtx = testRequire([], map, function() {});
+    var that = this;
+
+    // Define MockSettingsPanel
+    define('MockSettingsPanel', function() {
+      return function(options) {
+        return {
+          init: options.onInit,
+          beforeShow: options.onBeforeShow,
+          show: options.onShow,
+          beforeHide: options.onBeforeHide
+        };
+      };
+    });
+
+    // Define MockCallBarring
+    this.mockCallBarring = {
+      getAll: function(api) {},
+      set: function(api , setting, passcode) {
+        return new Promise(function (res, rej) {
+          res();
+        });
+      },
+      observe: function() {},
+      unobserve: function() {}
+    };
+    define('MockCallBarring', function() {
+      return that.mockCallBarring;
+    });
+
+    // Define MockPasscode
+    this.mockPasscode = {
+      init: function() {},
+      show: function() {}
+    };
+    define('MockPasscode', function() {
+      return function() {
+        return that.mockPasscode;
+      };
+    });
+
+    // Define MockToaster
+    this.mockToaster = {
+      showToast: function() {}
+    };
+    define('MockToaster', function() {
+      return that.mockToaster;
+    });
+
+    requireCtx(modules, function(CallBarringPanel) {
+      that.panel = CallBarringPanel();
+      done();
+    });
+
+    resetHTML();
+  });
+
```

Comment:
guess you're referring to other line? 
I don't see a problem on extra line at 135 between setup and test